platforms/core-execution/execution/src/main/java/org/gradle/internal/execution/FileCollectionSnapshotter.java

```diff
@@ -26,17 +27,17 @@
  */
 @ServiceScope({Scope.UserHome.class, Scope.BuildSession.class})
 public interface FileCollectionSnapshotter {
-    interface Result {
-        FileSystemSnapshot getSnapshot();
 
-        /**
-         * Whether any of the snapshot file collections is an archive tree backed by a file.
-         */
-        boolean containsArchiveTrees();
```

Comment:
💭 We don't need this ugly `Result` object anymore, since the only caller who was interested in archive trees now uses the visitor to find out if we bumped into any archive trees during snapshotting.